code/game/turfs/simulated/floor/lava_pool.dm

```diff
@@ -0,0 +1,59 @@
+/turf/simulated/floor/lava_pool
+		icon_state = "lava"
+		name = "lava pool"
+		desc = "A deadly pool of molten rock that incinerates almost anything on contact."
+		luminosity = 3
+		var/list/blacklist = list(/obj/item/clothing/suit/space/hardsuit/mining, /obj/item/asteroid/goliath_hide)
+		var/processing = 0
+
+/turf/simulated/floor/lava_pool/Entered(var/mob/living/carbon/C)
+	Incinerate(C) //To catch items before other things get processed
+	if(!istype(C))
+		return
+	if(!processing)
+		SSobj.processing |= src
+		processing = 1
+	if(istype(C, /mob/living/carbon/human))
+		if(check_suit(C))
+			return
+
+	apply_lava_damage(C)
+	C.emote("scream")
+	C.visible_message("<span class='warning'>[C] falls into [src]!   </span>", "<span class='userdanger'>You fall into [src]!   </span>", "<span class='warning'>You hear a splash followed by a howling scream! </span>")
+
+/turf/simulated/floor/lava_pool/proc/process()
+	var/mobpresent = 0
+	for(var/mob/living/carbon/human/H in src)
+		mobpresent = 1
+		if(check_suit(H))
+			continue
+
+		apply_lava_damage(H)//Defaults to 0 for message
+	if(!mobpresent)
+		SSobj.processing.Remove(src)
+		processing = 0
+
+/turf/simulated/floor/lava_pool/proc/check_suit(mob/living/carbon/human/H) // Checks for goliath plated suit
```

Comment:
This needs the first line of the proc to be
```
if(!istype(H))
    return 0
```
so that it doesn't runtime if somebody accidentally uses it on a non human.